clinica/iotools/converters/adni_to_bids/adni_utils.py

```diff
@@ -564,237 +510,194 @@ def remove_fields_duplicated(bids_fields):
     return [x for x in bids_fields if not (x in seen or seen_add(x))]
 
 
+def filter_subj_bids(df_files, location, bids_ids):
+    import clinica.iotools.bids_utils as bids
+
+    # Depending of the file that needs to be open, identify and
+    # do needed preprocessing on the column that contains the
+    # subjects ids
+    bids_ids = [x[8:] for x in bids_ids if "sub-ADNI" in x]
+    if location == "ADNIMERGE.csv":
+        df_files["RID"] = df_files["PTID"].apply(
+            lambda x: bids.remove_space_and_symbols(x)
+        )
+        df_files["RID"] = df_files["RID"].apply(lambda x: x[4:])
+        df_ret = df_files[df_files["RID"].isin([x[4:] for x in bids_ids])]
+    else:
+        df_files["RID"] = df_files["RID"].apply(lambda x: pad_id(x))
+        df_ret = df_files[df_files["RID"].isin([x[4:] for x in bids_ids])]
+    return df_ret
+
+
+def update_age(row):
+    """Update age with time passed since bl to current visit"""
+    from datetime import datetime
+
+    if row["session_id"] != "ses-M00":
+        examdate = datetime.strptime(row["EXAMDATE"], "%Y-%m-%d")
+        examdate_bl = datetime.strptime(row["EXAMDATE_bl"], "%Y-%m-%d")
+        delta = examdate - examdate_bl
+
+        updated_age = round(
+            float(row["AGE"]) + (delta.days / 365.25),
+            1,
+        )
+    else:
+        updated_age = row["AGE"]
+    return updated_age
+
+
+def pad_id(ref_id):
+    """Add leading zeros to the RID to keep à length of 4"""
+    rid = str(ref_id)
+    # Fill the rid with the needed number of zero
+    if 4 - len(rid) > 0:
+        zeros_to_add = 4 - len(rid)
+        rid = "0" * zeros_to_add + rid
+    return rid
```

Comment:
Same remark: if this is part of a higher processing loop on a Pandas dataframe, using Pandas string processing primitives would likely be more efficient.